src/material/core/option/optgroup.html

```diff
@@ -1,2 +1,2 @@
-<label class="mat-optgroup-label" [id]="_labelId">{{ label }} <ng-content></ng-content></label>
+<span class="mat-optgroup-label" aria-hidden="true" [id]="_labelId">{{ label }} <ng-content></ng-content></span>
```

Comment:
The same fix will have to be applied to the MDC-based option group in `src/material-experimental/mdc-core/option/optgroup.html`.

Reply:
Added the changes to `src/material-experimental/mdc-core/option/optgroup.html`. 

I described the a11y issues I saw in #21847 (with ChromeVox and with VoiceOver + Chrome). I believe removing the group labels from the a11y tree would stop the double-reading in these cases without impacting browsers that do not have this issue. I could provide some screen recordings if that would help.